Lib/site.py

```diff
@@ -523,8 +523,9 @@ def register_readline():
             pass
 
         def write_history():
+            import _pyrepl
             try:
-                if os.getenv("PYTHON_BASIC_REPL"):
+                if os.getenv("PYTHON_BASIC_REPL") or not _pyrepl._CAN_USE_PYREPL:
```

Comment:
question: This is working, I presume, because we're hitting the `except` block in `__main__` when we have a `dumb` TERM setting? I was initially a bit confused because I wasn't seeing why an extra conditional on a setting for a win32 check would fix this

Reply:
`_pyrepl._CAN_USE_PYREPL` is set to False if pyrepl fails at startup.